src/detect-transform-base64.c

```diff
@@ -0,0 +1,335 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Jeff Lucovsky <jlucovsky@oisf.net>
+ *
+ * Implements the from_base64 transformation keyword
+ */
+
+#include "suricata-common.h"
+
+#include "detect.h"
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-byte.h"
+
+#include "rust.h"
+
+#include "detect-transform-base64.h"
+
+#include "util-base64.h"
+#include "util-unittest.h"
+#include "util-print.h"
+
+static int DetectTransformFromBase64DecodeSetup(DetectEngineCtx *, Signature *, const char *);
+static void DetectTransformFromBase64DecodeFree(DetectEngineCtx *, void *);
+#ifdef UNITTESTS
+static void DetectTransformFromBase64DecodeRegisterTests(void);
+#endif
+static void TransformFromBase64Decode(InspectionBuffer *buffer, void *options);
+
+#define DETECT_TRANSFORM_FROM_BASE64_MODE_DEFAULT (uint8_t) ModeRFC4648
+
+void DetectTransformFromBase64DecodeRegister(void)
+{
+    sigmatch_table[DETECT_TRANSFORM_FROM_BASE64].name = "from_base64";
+    sigmatch_table[DETECT_TRANSFORM_FROM_BASE64].desc = "convert the base64 decode of the buffer";
+    sigmatch_table[DETECT_TRANSFORM_FROM_BASE64].url = "/rules/transforms.html#from_base64";
+    sigmatch_table[DETECT_TRANSFORM_FROM_BASE64].Setup = DetectTransformFromBase64DecodeSetup;
+    sigmatch_table[DETECT_TRANSFORM_FROM_BASE64].Transform = TransformFromBase64Decode;
+    sigmatch_table[DETECT_TRANSFORM_FROM_BASE64].Free = DetectTransformFromBase64DecodeFree;
+#ifdef UNITTESTS
+    sigmatch_table[DETECT_TRANSFORM_FROM_BASE64].RegisterTests =
+            DetectTransformFromBase64DecodeRegisterTests;
+#endif
+    sigmatch_table[DETECT_TRANSFORM_FROM_BASE64].flags |= SIGMATCH_OPTIONAL_OPT;
+}
+
+static void DetectTransformFromBase64DecodeFree(DetectEngineCtx *de_ctx, void *ptr)
+{
+    ScTransformBase64Free(ptr);
+}
+
+static DetectTransformFromBase64Data *DetectTransformFromBase64DecodeParse(const char *str)
+{
+    DetectTransformFromBase64Data *tbd = ScTransformBase64Parse(str);
+    if (tbd == NULL) {
+        SCLogError("invalid transform_base64 values");
+    }
+    return tbd;
+}
+
+/**
+ *  \internal
+ *  \brief Base64 decode the input buffer
+ *  \param det_ctx detection engine ctx
+ *  \param s signature
+ *  \param opts_str transform options, if any
+ *  \retval 0 No decode
+ *  \retval >0 Decoded byte count
+ */
+static int DetectTransformFromBase64DecodeSetup(
+        DetectEngineCtx *de_ctx, Signature *s, const char *opts_str)
+{
+    int r = -1;
+
+    SCEnter();
+
+    DetectTransformFromBase64Data *b64d = DetectTransformFromBase64DecodeParse(opts_str);
+    if (b64d == NULL)
+        SCReturnInt(r);
+
+    if (b64d->flags & DETECT_TRANSFORM_BASE64_FLAG_OFFSET_VAR) {
+        SCLogError("offset value must be a value, not a variable name");
```

Comment:
Is this a planned addition ?

Reply:
Yes -- https://redmine.openinfosecfoundation.org/issues/6807

Reply:
Interesting, keep me posted on this : I wonder if we could also use it for generic integer keywords